gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -306,6 +312,18 @@ public void launchJob(JobListener jobListener) throws JobException {
     }
   }
 
+  private void executeUnfinishedCommitSequences(String jobName) throws IOException {
+    Preconditions.checkState(this.jobContext.getCommitSequenceStore().isPresent());
+
+    for (String datasetUrn : this.jobContext.getCommitSequenceStore().get().get(jobName)) {
+      Optional<CommitSequence> commitSequence = this.jobContext.getCommitSequenceStore().get().get(jobName, datasetUrn);
+      if (commitSequence.isPresent()) {
+        commitSequence.get().execute();
+      }
+      this.jobContext.getCommitSequenceStore().get().delete(jobName, datasetUrn);
```

Comment:
Seems you can have a local variable for `this.jobContext.getCommitSequenceStore()`.